src/pages/Home.jsx

```diff
@@ -1,12 +1,42 @@
-import React from "react";
+import React, { useState } from "react";
+import { Country } from "../components/Country";
+import Data from "../assets/CountriesData.json";
 
 const Home = () => {
-    return (
-        // TODO: Home page
-        // Render Country component (components/Country.jsx) for each country
-        // Take data from (assets/CountriesData.json)
-        <div>Home</div>
+ 
+  const [countries, setCountries] = useState(Data);
+
+
+  const onSearch = (e) => {
+    const searchValue = e.target.value.toLowerCase(); 
+    const filteredData = Data.filter((country) =>
+      country.name.toLowerCase().includes(searchValue) 
     );
+    console.log(filteredData);
+    setCountries(filteredData); 
+  };
+
+  return (
+    <div>
+      <input
+        type="text"
+        placeholder="Search by country name..."
+        onChange={onSearch}
+      />
+      <div className="countries-grid">
+        {countries.map((country) => (
+          <Country
+            key={country.name}
```

Comment:
You can add another props to you Card component like this : 

```js
  return (
    <div {...props}>
      <img src={flag} alt={`${name} flag`} width="100" />
      <h2>{name}</h2>
      <p>Population: {population.toLocaleString()}</p>
      <p>Region: {region}</p>
      <p>Capital: {capital}</p>
    </div>
  );
};

```

Then all you need to do it call you component like this:

```js
<Country
     //Your props goes in here
      {...country}}
>
```